docs/src/main/docs/health/01_health.adoc

```diff
@@ -20,71 +20,192 @@
+ inline with a lambda expression or you can reference a method with the double
+ colon operator `::`.
+
+[source,java]
+.Health check with a lambda expression
 ----
-    <dependency>
-        <groupId>io.helidon.health</groupId>
-        <artifactId>helidon-health-checks</artifactId>
-    </dependency>
+HealthCheck hc = () -> HealthCheckResponse
+        .named("exampleHealthCheck")
+        .up()
+        .build();
 ----
 
+[source,java]
+.Health check with method reference
+----
+HealthCheckResponse exampleHealthCheck(){
+    return HealthCheckResponse
+        .named("exampleHealthCheck")
+        .up()
+        .build();
+}
+HealthCheck hc = this::exampleHealthCheck;
+----
+
+`HealthSupport` is a WebServer service that contains a collection of
+ registered `HealthCheck` instances. When queried, it invokes the registered
+ health check and returns a response with a status code representing the overall
+ state of the application.
+
+[cols="1,5",role="flex, sm7"]
+.Health status codes
+|=======
+| `200` | All health checks succeeded
+| `503` | At least one health check failed
+| `500` | An error occurred while reporting the health
+|=======
 
-=== Adding health checks to your application
-To enable health checks, register them with the web server.
+The HTTP response also contains a JSON payload that describes the statuses for
+ all health checks.
 
 [source,java]
+.Create the health support service
 ----
-import io.helidon.metrics.MetricsSupport;
-//...
-HealthSupport health = HealthSuport.builder()
-    .add(HealthChecks.healthChecks())               // <1>
+HealthSupport health = HealthSupport.builder()
+    .add(this::exampleHealthCheck)
+    .build();
+----
+
+NOTE: Health check requires the `JSON-P` support to be enabled, see the example
+ below.
+
+=== Example
+
+[source,java]
+.Register a custom health check
+----
+HealthSupport health = HealthSupport.builder()
     .add(() -> HealthCheckResponse.named("exampleHealthCheck")
                  .up()
                  .withData("time", System.currentTimeMillis())
-                 .build())                          // <2>
+                 .build()) // <1>
     .build();
 
 Routing.builder()
-        .register(health)                           // <3>
-        .register("/hello", (req, res) -> res.send("HelloWorld"))
+        .register(JsonSupport.create()) // <2>
+        .register(health) // <3>
         .build();
 ----
-<1> Add built-in health checks (requires the `helidon-health-checks` dependency)
-<2> Add a custom health check. This example returns `UP` and current time
-<3> Register health support with web server routing (adds the `/health` endpoint)
+<1> Add a custom health check. This example returns `UP` and current time
+<2> Enable support for `JSON`
+<3> Register health support with web server routing (adds the `/health`
+ endpoint)
 
-=== Accessing the health check endpoint
+TIP: Remember to balance the amount of collected information in order to avoid
```

Comment:
Might not need "Remember to" if we are striving for brevity. 

Also, maybe do one of the following:
1. Change "balance" to "be selective about." Something is balanced with or against something else.
2. If we like "balance" then maybe change the sentence to "Balance collecting a lot of information with the need to avoid overloading the application and overwhelming users."

Option 2 gets pretty long-winded IMO but both communicate the idea, I think.

Reply:
Option 2 is nice, I will go for it.